app/src/main/java/org/lineageos/selfie/MainActivity.kt

```diff
@@ -587,67 +586,65 @@ class MainActivity : AppCompatActivity() {
         val outputOptions = StorageUtils.getVideoMediaStoreOutputOptions(contentResolver, location)
 
         // Play shutter sound
-        if (cameraSoundsUtils.playStartVideoRecording()) {
-            // Delay startRecording() by 500ms to avoid recording shutter sound
-            delay(500)
-        }
-
-        // Start recording
-        recording = cameraController.startRecording(
-            outputOptions,
-            audioConfig,
-            cameraExecutor
-        ) {
-            val updateRecordingStatus = { enabled: Boolean, duration: Long ->
-                // Hide mode buttons
-                photoModeButton.isInvisible = enabled
-                videoModeButton.isInvisible = enabled
-                qrModeButton.isInvisible = enabled
-
-                // Update duration text and visibility state
-                videoDuration.text = TimeUtils.convertNanosToString(duration)
-                videoDuration.isVisible = enabled
-
-                // Update video recording pause/resume button visibility state
-                if (duration == 0L) {
-                    flipCameraButton.isInvisible = enabled
-                    videoRecordingStateButton.isVisible = enabled
+        val delayTime = if (cameraSoundsUtils.playStartVideoRecording()) 500L else 0L
+
+        shutterButton.postDelayed({
```

Comment:
wouldn't we want handler.postDelayed() here?

Reply:
Done